api/client.ts

```diff
@@ -155,7 +155,7 @@ function connectAndLogin(url: string, credentials, options) {
       const conn = await juju.login(credentials);
       resolve({ conn, logout: juju.logout.bind(juju) });
     } catch (error) {
-      if (!juju.isRedirectionError(error)) {
+      if (!juju || !juju.isRedirectionError(error)) {
```

Comment:
This is the fix for the connection error, the rest is to add the schema to the source tree.